src/components/Admin/Categories.js

```diff
@@ -32,17 +26,52 @@ class Categories extends Component {
   }
 
   handleOk = (e) => {
+    this.props.addCategory(this.state.newCategoryName);
     this.setState({
       visible: false,
+      newCategoryName: ''
     });
   }
 
   handleCancel = (e) => {
     this.setState({
       visible: false,
+      newCategoryName: ''
+    });
+  }
+
+  updateShowModal = () => {
+    this.setState({
+      visibleUpdateModal: true,
     });
   }
 
+  handleOkUpdate = (id) => {
+    this.props.updateCategory(this.state.updateCategoryName,id);
+    this.setState({
+      visibleUpdateModal: false,
+      newCategoryName: ''
+    });
+  }
+
+  handleCancelUpdate = (e) => {
+    this.setState({
+      visibleUpdateModal: false,
+      newCategoryName: ''
+    });
+  }
+
+  handleDelete = (id) => {
+    this.props.deleteCategory(id);
+  }
+  componentDidUpdate(){
+    this.props.fetchCategory();
+  }
+  componentWillMount(){
```

Comment:
Change this to `componentDidMount` . When fetching data from the server for the first time always use `componentDidMount`